flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBKeyedStateBackendBuilder.java

```diff
@@ -500,9 +487,18 @@ private static void checkAndCreateDirectory(File directory) throws IOException {
                             lastCompletedCheckpointId,
                             numberOfTransferingThreads);
         } else {
-            checkpointSnapshotStrategy = savepointSnapshotStrategy;
+            checkpointSnapshotStrategy =
+                    new RocksFullSnapshotStrategy<>(
+                            db,
+                            rocksDBResourceGuard,
+                            keySerializerProvider.currentSchemaSerializer(),
+                            kvStateInformation,
+                            keyGroupRange,
+                            keyGroupPrefixBytes,
+                            localRecoveryConfig,
+                            keyGroupCompressionDecorator);
         }
-        return new SnapshotStrategy<>(checkpointSnapshotStrategy, savepointSnapshotStrategy);
+        return new SnapshotStrategy<>(checkpointSnapshotStrategy);
```

Comment:
We can get rid of the `SnapshotStrategy` wrapper.

Reply:
Good catch! Removing this.